src/fidesops/service/connectors/saas_connector.py

```diff
@@ -310,64 +310,33 @@ def process_response_data(
 
         return rows
 
-    def get_masking_request_from_config(
-        self, collection_name: str
-    ) -> Optional[SaaSRequest]:
-        """Get the configured SaaSRequest for use in masking.
-        An update request is preferred, but we can use a gdpr delete endpoint or delete endpoint if not MASKING_STRICT.
-        """
-        requests: Dict[
-            Literal["read", "update", "delete"], SaaSRequest
-        ] = self.endpoints[collection_name].requests
-
-        update: Optional[SaaSRequest] = requests.get("update")
-        gdpr_delete: Optional[SaaSRequest] = None
-        delete: Optional[SaaSRequest] = None
-
-        if not config.execution.MASKING_STRICT:
-            gdpr_delete = self.saas_config.data_protection_request
-            delete = requests.get("delete")
-
-        try:
-            # Return first viable option
-            action_type: str = next(
-                action
-                for action in [
-                    "update" if update else None,
-                    "data_protection_request" if gdpr_delete else None,
-                    "delete" if delete else None,
-                ]
-                if action
-            )
-            logger.info(
-                f"Selecting '{action_type}' action to perform masking request for '{collection_name}' collection."
-            )
-            return next(request for request in [update, gdpr_delete, delete] if request)
-        except StopIteration:
-            return None
-
     def mask_data(
         self,
         node: TraversalNode,
         policy: Policy,
         privacy_request: PrivacyRequest,
         rows: List[Row],
     ) -> int:
-        """Execute a masking request. Return the number of rows that have been updated"""
+        """Execute a masking request. Return the number of rows that have been updated."""
+
         query_config = self.query_config(node)
-        if not query_config.masking_request:
-            raise Exception(
+        masking_request = query_config.get_masking_request()
+        if not masking_request:
+            logger.warning(
```

Comment:
I changed this to a warning instead of an exception because I wanted to avoid retrying a task that will fail again.

Reply:
That's a good point, reverting back